src/NotebookOutputViewer/NotebookOutputViewer.tsx

```diff
@@ -0,0 +1,74 @@
+import { createImmutableOutput, JSONObject, OnDiskOutput } from "@nteract/commutable";
+import { KernelOutputError, Output, StreamText } from "@nteract/outputs";
+import { ContentRef } from "@nteract/types";
+import "bootstrap/dist/css/bootstrap.css";
+import postRobot from "post-robot";
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import "../../externals/iframeResizer.contentWindow.min.js"; // Required for iFrameResizer to work
+import "../Explorer/Notebook/NotebookRenderer/base.css";
+import "../Explorer/Notebook/NotebookRenderer/default.css";
+import { TransformMedia } from "./TransformMedia";
+
+export interface OutputsProps {
+  id: string;
+  contentRef: ContentRef;
+  hidden: boolean;
+  expanded: boolean;
+  outputs: OnDiskOutput[];
+  onMetadataChange: (metadata: JSONObject, mediaType: string, index?: number) => void;
+}
+
+const onInit = async () => {
+  postRobot.on(
+    "props",
+    {
+      window: window.parent,
+      domain: window.location.origin,
+    },
+    (event) => {
+      // Typescript definition for event is wrong. So read props by casting to <any>
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const data = (event as any).data;
+      const props: OutputsProps = {
+        id: data.id,
+        contentRef: data.contentRef,
+        hidden: data.hidden,
+        expanded: data.expanded,
+        outputs: data.outputs,
+        onMetadataChange: data.onMetadataChange,
+      };
```

Comment:
nit: can't we just use `event.data` as-is or we actually need a subset of the fields?
```suggestion
      const props: OutputsProps = (event as any).data;
```